File: user-service/src/main/java/socialapp/userservice/service/impl/AddressServiceImpl.java
package socialapp.userservice.service.impl;

import lombok.RequiredArgsConstructor;
import org.springframework.stereotype.Service;
import socialapp.userservice.common.exception.EntityNotFoundException;
import socialapp.userservice.common.mapper.AddressMapper;
import socialapp.userservice.model.dto.AddressDto;
import socialapp.userservice.model.entity.Address;
import socialapp.userservice.repository.AddressRepository;
import socialapp.userservice.service.AddressService;

@Service
@RequiredArgsConstructor
public class AddressServiceImpl implements AddressService {

    private final AddressRepository addressRepository;
    private final AddressMapper addressMapper;

    @Override
    public Address save(AddressDto addressDto) {
        return addressRepository.save(addressMapper.toModel(addressDto));
    }

    @Override
    public void update(AddressDto addressDto, Long id) {
        var address = findById(id);

        addressMapper.update(address, addressDto);

        addressRepository.save(address);
    }

    private Address findById(Long id) {
        return addressRepository.findById(id)
                .orElseThrow(() -> new EntityNotFoundException(Address.class, id));
    }

}

File: user-service/src/main/kotlin/socialapp/ktuserservice/service/impl/AddressServiceImpl.kt
package socialapp.ktuserservice.service.impl

import org.springframework.stereotype.Service
import socialapp.ktuserservice.common.mapper.AddressMapper
import socialapp.ktuserservice.model.dto.AddressDto
import socialapp.ktuserservice.model.entity.Address
import socialapp.ktuserservice.repository.AddressRepository
import socialapp.ktuserservice.service.AddressService

@Service
class AddressServiceImpl(
    private var addressRepository: AddressRepository,
    private var addressMapper: AddressMapper
) : AddressService {

    override fun save(addressDto: AddressDto): Address =
        addressRepository.save(addressMapper.toModel(addressDto));

    override fun update(addressDto: AddressDto,address: Address) {
        addressMapper.update(address, addressDto);
        addressRepository.save(address);
    }

}